Postgres docker DB

javipacheco opened the pull request:
Docker compose for Postgres DB

javipacheco committed:
Postgres docker DB

SuliAragon committed:
Update Readme.md

Update of the README for the installation and use of Docker.

Montagon commented on README.md:
```diff
@@ -56,11 +56,53 @@ mvn spring-boot:run
+
+## Docker Compose Configuration
+
+The project use Docker Compose to spin up a PostgreSQL container, here is a summary of the configuration form .yml: 
+
+    
+    services:
+    brain-ai-vector-store:
+    container_name: board-game-training-db
+    image: "pgvector/pgvector:pg17"
+    volumes:
+    - ./postgres-dev-data/17/data:/var/lib/postgresql/data
+    ports:
+      - "5490:5432"
+      expose:
+      - "5490"
+      healthcheck:
+      test: [ "CMD", "pg_isready", "-U", "postgres" ]
+      interval: 2s
+      timeout: 2s
+      retries: 5
+      restart: always
+      environment:
+      POSTGRES_DB: board-game-training-db
+      POSTGRES_USER: ${BOARD_GAME_TRAINING_DB_USERNAME:-postgres}
+      POSTGRES_PASSWORD: ${BOARD_GAME_TRAINING_DB_PASSWORD:-postgres}
+
```
I think it's better to link to the file instead of writing here, if changed, you need to always update here, better to maintain only one file of truth

javipacheco commented on README.md:
```diff
@@ -31,9 +31,9 @@ spring.jpa.database-platform=org.hibernate.dialect.PostgreSQLDialect
 
 #Database properties
-spring.datasource.url=jdbc:postgresql://localhost:5432/yourDatabaseName
+spring.datasource.url=jdbc:postgresql://localhost:5490/board-game-training-db
```
As @Montagon said below, this information is not necessary right now, we can remove it from the README 

We only can remember to the user if he need to change something related to the database, he should change it in the `application.properties` file

SuliAragon committed:
Changes, in pom.xml, test, applications.properties and BoardGameTrainingApplication

- I modified the Readme file to make it more user-friendly by modifying line 34, which indicates where to change the database name.
- At the last minute, I discovered one of the errors in the pom.xml file: the JUnit dependency was not found in the file.
- I added an @Bean tag with a RestTemplate to BoardGameTrainingApplication in the Java root folder.
- I generated the first test correctly, and the backend is now ready to generate several tests and verify that everything is working correctly.
- The error found yesterday, on the other computer, is simply in the database generation. After testing, it's best to create a database in PostgreSQL that maintains the same name and password to validate it as valid.

SuliAragon committed:
Changes made to pom.xml, tests, controller and service

Changes made to pom.xml, installing dependencies for mockito-core

Changes made to tests, with several tests performed, simulating different situations

A create method was created for the service and the controller and tested with Postman to verify its functionality

SuliAragon committed:
Update GameController.java

SuliAragon committed:
Update pom.xml

- Removing duplicate plugins
- Installing Maven
- Upgrading from LASTEST to specific versions of Mockito and JUnit

SuliAragon committed:
[ADD] Endpoint for delete and collection.json

Both GameController and GameService have been updated to support deleting a game.

Added the exported Postman collection file.

Montagon commented on README.md:
```diff
@@ -56,11 +56,53 @@ mvn spring-boot:run
+
+## Docker Compose Configuration
+
+The project use Docker Compose to spin up a PostgreSQL container, here is a summary of the configuration form .yml: 
+
+    
+    services:
+    brain-ai-vector-store:
+    container_name: board-game-training-db
+    image: "pgvector/pgvector:pg17"
+    volumes:
+    - ./postgres-dev-data/17/data:/var/lib/postgresql/data
+    ports:
+      - "5490:5432"
+      expose:
+      - "5490"
+      healthcheck:
+      test: [ "CMD", "pg_isready", "-U", "postgres" ]
+      interval: 2s
+      timeout: 2s
+      retries: 5
+      restart: always
+      environment:
+      POSTGRES_DB: board-game-training-db
+      POSTGRES_USER: ${BOARD_GAME_TRAINING_DB_USERNAME:-postgres}
+      POSTGRES_PASSWORD: ${BOARD_GAME_TRAINING_DB_PASSWORD:-postgres}
```
```suggestion
The project uses Docker Compose to spin up a PostgreSQL container, please check the file to see the details: [docker-compose.yml](docker-compose.yml).
```

Montagon commented on src/test/java/com/es/boardGameTraining/BoardGameTrainingApplicationTests.java:
```diff
@@ -1,13 +1,152 @@
+		 g.setType(type);
+		 g.setYear(year);
+
+		 assertEquals(bggID, g.getBggId()); // comprueba que el parametro que recibe es igual que el contenido de bggID
```
better in English

Montagon commented on README.md:
```diff
@@ -56,11 +56,53 @@ mvn spring-boot:run
+
+### 5 Notes on Docker and Batabase
+
+- If you wish to modify the database name, username, or password, you can do so by changing the values in the `docker-compose.yml` and the `application.properties` file. 
```
```suggestion
- If you wish to modify the database name, username, or password, you can do so by changing the values in the [`docker-compose.yml`](docker-compose.yml) and the [`application.properties`](src/main/resources/application.properties) file. 
```

Montagon commented on src/main/java/com/es/boardGameTraining/BoardGameTrainingApplication.java:
```diff
@@ -10,4 +12,9 @@ public static void main(String[] args) {
 		SpringApplication.run(BoardGameTrainingApplication.class, args);
 	}
 
+	@Bean // Creacion mas explicita que el @Autowired
```
comments better in English

Montagon commented on src/test/java/com/es/boardGameTraining/BoardGameTrainingApplicationTests.java:
```diff
@@ -1,13 +1,152 @@
+	}
+
+
+	// verifica si las propiedades de la clase Game tienen valores predertminados esperados, que en este caso son 0
```
The same, better in English

Montagon reviewed (changes requested):
Just minor comments, but prefered to be in the code.

SuliAragon committed:
all changes made

all changed made, all changes made to the different files, including comments in English

Montagon commented on src/main/resources/application.properties:
```diff
@@ -1 +1,18 @@
+#Database properties
+spring.datasource.url=jdbc:postgresql://localhost:5432/board-game-training-db
+spring.datasource.username=postgres
+spring.datasource.password=1234
```
the password is 1234 or postgres?

SuliAragon replied:
I changed it so that it makes sense with the original project, but in principle the password is consistent with the database created in Postgree, but I changed it so that it makes sense

SuliAragon committed:
made change password
